File: Graph/ScriptTemplate/Not/CGNot.js
/**
 * @file CGNot.js
 * @author
 * @date 2021/8/15
 * @brief CGOr.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const {BaseNode} = require('./BaseNode');
const Amaz = effect.Amaz;

class CGNot extends BaseNode {
  constructor() {
    super();
  }

  getOutput() {
    return !this.inputs[0]();
  }
}

exports.CGNot = CGNot;

File: Graph/ScriptTemplate/Not/CGNot.lua
local CGNot = CGNot or {}
CGNot.__index = CGNot

function CGNot.new()
    local self = setmetatable({}, CGNot)
    self.inputs = {}
    return self
end

function CGNot:setInput(index, func)
    self.inputs[index] = func
end

function CGNot:getOutput(index)
    return not self.inputs[0]()
end

return CGNot
